### PR TITLE
2.x Extend available TLS encrypt methods in CakeSocket

### PR DESCRIPTION
**Pull request by mstramek-CKPL**
Proposed change extend list of available encryption methods in CakeSocket with two new ones: TLSv1_1 and TLSv1_2. Moreover it added the possibility to choose custom TLS version by config option.

### CODE REVIEW

**Comment by markstory** [comment]
Your changes make [tests fail](https://travis-ci.org/cakephp/cakephp/jobs/201810255) as the constants you're using do not exist in some of the PHP versions we support.

File: lib/Cake/Network/CakeSocket.php [hunk 1]
```diff
@@ -44,7 +44,8 @@ class CakeSocket {
 		'host' => 'localhost',
 		'protocol' => 'tcp',
 		'port' => 80,
-		'timeout' => 30
+		'timeout' => 30,
+		'cryptoType' => 'tls'
```
Comment (markstory) [inline lib/Cake/Network/CakeSocket.php]: What other values would go in here? There aren't any tests or documentation for this config option.

**Comment by ravage84** [comment]
@m-stramek Related links would help reviewing this.

**Comment by ravage84** [comment]
Refers: #10230

File: lib/Cake/Network/CakeSocket.php [hunk 2]
```diff
@@ -93,10 +94,14 @@ class CakeSocket {
...
 		'tls_client' => STREAM_CRYPTO_METHOD_TLS_CLIENT,
+		'tlsv1_1_client' => STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT,
+		'tlsv1_2_client' => STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT,
 		'sslv2_server' => STREAM_CRYPTO_METHOD_SSLv2_SERVER,
 		'sslv3_server' => STREAM_CRYPTO_METHOD_SSLv3_SERVER,
 		'sslv23_server' => STREAM_CRYPTO_METHOD_SSLv23_SERVER,
-		'tls_server' => STREAM_CRYPTO_METHOD_TLS_SERVER
+		'tls_server' => STREAM_CRYPTO_METHOD_TLS_SERVER,
+		'tlsv1_1_server' => STREAM_CRYPTO_METHOD_TLSv1_1_SERVER,
+		'tlsv1_2_server' => STREAM_CRYPTO_METHOD_TLSv1_2_SERVER
```
Comment (markstory) [inline lib/Cake/Network/CakeSocket.php]: These constants are not defined in PHP5.5 and lower. You should use `constant()` to read their values, or conditionally set the keys based on the constants existing.

**Comment by markstory** [comment]
Replacing this with #10482 so I can get tests to pass.

**Pull request closed without merging**